src/main/java/net/fabricmc/loom/task/RemapSourcesJarTask.java

```diff
@@ -28,29 +28,28 @@
 
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.InputFile;
-import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.jvm.tasks.Jar;
 
 import net.fabricmc.loom.util.SourceRemapper;
 
-public class RemapSourcesJarTask extends AbstractLoomTask {
+public class RemapSourcesJarTask extends Jar {
 	private Object input;
-	private Object output;
 	private String direction = "intermediary";
 
 	@TaskAction
 	public void remap() throws Exception {
-		SourceRemapper.remapSources(getProject(), getInput(), getOutput(), direction.equals("named"));
+		SourceRemapper.remapSources(getProject(), getInput(), getArchivePath(), direction.equals("named"));
 	}
 
 	@InputFile
 	public File getInput() {
 		return getProject().file(input);
 	}
 
-	@OutputFile
+	@Deprecated // Use getArchivePath
```

Comment:
Use javadocs, for example 
```
/**
 * @deprecated  As of release 1.3, replaced by {@link #getPreferredSize()}
 */
@Deprecated public Dimension preferredSize() {
return getPreferredSize();
}
```

Reply:
will be removed in.0.2.7, and old one has no jd so i don't think this is necessary

Reply:
alright